### ISSUE TITLE
Typo in paper equation

### ISSUE THREAD

**Issue by a-jahani** [hufu6371/DORN#20]
When I was reading the paper a long time ago I think there is a potentially important typo there. Please confirm if I am correct or wrong.

In equation (2) the second term should be:
`SIGMA (1- log(P) )` instead of  `SIGMA(log(1-P))`

Let's assume if a true depth is k (in uniform distribution),  you want:
```
P0 = 1
P1 = 1 
.
.
.
P(K-1) = 1
```

and

```
PK = 0
P(K+1) = 0
.
.
PN = 0

```
This means there is a typo because your equation will give you` log(0) for Pi when i>k-1` which will be infinity in LOSS.

**Comment by pobingwanghai** [comment]
@a-jahani I also get a question about equation 2 and 3, the k index seems not right. In equation3, the Pk is computed from 2k and 2k+1, then it means that the equation2 is wrong. The first part of sum operation should be computed on 2k while the second part is computed on 2k+1, not the 0-K and K-2K.

**Comment by a-jahani** [comment]
Equation 3 typo:
![image](https://user-images.githubusercontent.com/13435607/49554909-07e7ba80-f8bb-11e8-947e-330c33edabc3.png)

'k' should be '2k'. But what they mean is clear and won't affect the loss:
 
Your network outputs 2k classes. Then there is a softmax over those pairs of probability (2K class ; k intervals) to make those pairs sum to one
![image](https://user-images.githubusercontent.com/13435607/49554448-7f1c4f00-f8b9-11e8-9e92-aeb2462433d8.png) 

Now they use those k normalized probabilities to calculate equation 2. 

About equation 2 typo. I was wrong. Equation 2 is correct. It's a simple negative log likelihood.

I also have an idea of making loss better but unfortunately the training is not public and I don't have time to reproduce the training stage. The concept of ordinal here is not fully exploited. If anyone were able to reproduce their training, would be glad to shoot me an email and chat.

**Comment by pobingwanghai** [comment]
@a-jahani I have failed to reproduce the training by pytorch on the NYU dataset and I don't recommend anyone to reproduce the training process.
The results are not as good as shown in the paper. And the parameter setting is actually different from the Eigen in NYU and KITTI. The image size is strange which is 257*353 and the sub-sampling scale is 1.6 which is different from the scale 2 in Eigen's paper. I got no idea how to evaluate this kind of setting with the standard 228*304 image with scale 2. So I just use the standard Eigen's setting for the training which may lead to a worse performance. In addition, for KITTI, I don't know how to crop 385*513 from the 375*1241.

**Comment by a-jahani** [comment]
The scaling shouldn't be an issue. Do u mind if u share your code so far? U can drop me an email ("jahaniam@ualberta.ca") and I'll have a look.

For evaluation no matter what is the output size you resize it to the original size and evaluate in that scale. The inference code of DORN they crop the image and feed it multiple times and at the end, they merge all those crops to get the final output. That's why if you look closely you can see some seams(vertical lines).
